src/NHibernate/Type/AbstractDateTimeType.cs

```diff
@@ -176,7 +176,7 @@ public override object FromStringValue(string xml)
 		public override object DefaultValue => BaseDateValue;
 
 		/// <inheritdoc />
-		public override string ObjectToSQLString(object value, Dialect.Dialect dialect) =>
-			"'" + (DateTime) value + "'";
+		public override string ObjectToSQLString(object value, Dialect.Dialect dialect)
+			=> dialect.ToStringLiteral(((DateTime) value).ToString(), SqlTypeFactory.GetAnsiString(50));
```

Comment:
That is still far from being a correct datetime representation for many SQL vendors and cultures. But at least, it does no more allow SQL injections. Granted, injecting through a very custom culture is a hard to exploit attack vector, even more with the unlikeliness of the vulnerable features to be used by applications. But after all, who knows how corrupted a computer culture might be?